src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerJsonNetWithoutBinder.cs

```diff
@@ -0,0 +1,242 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.FlowAnalysis.Analysis.PropertySetAnalysis;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.ValueContentAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.NetCore.Analyzers.Security.Helpers;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    /// <summary>
+    /// For detecting deserialization with <see cref="T:Newtonsoft.Json.JsonSerializer"/> when its TypeNameHandling != None
+    /// and its Binder/SerializationBinder is null.
+    /// </summary>
+    [SuppressMessage("Documentation", "CA1200:Avoid using cref tags with a prefix", Justification = "The comment references a type that is not referenced by this compilation.")]
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    internal class DoNotUseInsecureDeserializerJsonNetWithoutBinder : DiagnosticAnalyzer
+    {
+        // TODO paulming: Help links URLs.
+        internal static readonly DiagnosticDescriptor DefinitelyInsecureSerializer =
+            SecurityHelpers.CreateDiagnosticDescriptor(
+                "CA2329",
+                nameof(MicrosoftNetCoreAnalyzersResources.JsonNetInsecureSerializerTitle),
+                nameof(MicrosoftNetCoreAnalyzersResources.JsonNetInsecureSerializerMessage),
+                isEnabledByDefault: false,
+                helpLinkUri: null);
```

Comment:
`customTags: WellKnownCustomTagExtensions.DataflowAndTelementry`

Reply:
By the way, I am planning to enhance our ruleset generation to also generate DFA specific rulesets so users can bulk configure DFA rules. I will be using the DataFlow custom tag to generate such a ruleset:

![image](https://user-images.githubusercontent.com/10605811/61963140-da519d80-af7f-11e9-83f5-4b6733aea174.png)